### PR TITLE
build foundation-e2e image per branch

### PR DESCRIPTION
**Pull request by elgnay**

### CODE REVIEW

File: ci-operator/config/stolostron/foundation-e2e/stolostron-foundation-e2e-main.yaml [hunk 1]
```diff
@@ -44,14 +44,6 @@ resources:
       memory: 200Mi
 test_binary_build_commands: make build GO_BUILD_FLAGS:='-race' --warn-undefined-variables
 tests:
-- as: pr-image-mirror
```
Comment (zhujian7) [inline ci-operator/config/stolostron/foundation-e2e/stolostron-foundation-e2e-main.yaml:47]: why delete this?

Comment (elgnay) [reply ci-operator/config/stolostron/foundation-e2e/stolostron-foundation-e2e-main.yaml:47]: This repo does not need images like `2.8.0-PR2-f9c58a786ded99a9e2ec8b8b08cc6fe397184d81`

**Pull request merged**